### PR TITLE
User interface set up for milestone 2

### PR DESCRIPTION
**Pull request by KabohaJeanMark**
- Add Command Line Interface to welcome users and get their names and assign playing token (X or O).
- Loop through and ask users for valid input of numbers on the board.
- Clear terminal and terminal lines to always display the tic tac toe board while asking users for input.
- After looping for a max of 9 possible times, a random decision is made either win for a user or a tie (Sidenote: Actual game logic to be worked on in next milestone).
- A visual walkthrough of the gameplay can be seen on the README.md

### CODE REVIEW

**Review by DeliceLydia — CHANGES_REQUESTED** [review]
# Status: Change Requested image

Great job @KabohaJeanMark and @chasscepts you did an amazing job👏👏 so far

- Your user interface is so amazing I liked it
- Your README is very descriptive
- Your PR is professional

However, you still have some small changes to make 
- User input is not validated it can accept an empty name try to fix that and submit for another code review
# Happy Coding image

Best regards,

Lydia Ingabire

**Comment by KabohaJeanMark** [comment]
Thanks, 👍🏾  @DeliceLydia for the quick, concise and precise feedback.
@chasscepts and I have worked on it to improve the project with input validation for the user name :muscle:

**Review by BrianSammit — CHANGES_REQUESTED** [review]
CHANGES REQUIRED :red_circle: 
------------
Hello, @KabohaJeanMark @chasscepts  You are doing an amazing job on this milestone. I really like how the interface of the game looks :muscle: :1st_place_medal: 
There is one a change to make to complete this milestone :hammer_and_wrench: see it below :eyes: :point_down: 
--------------

- [ ] A loop is needed in this game. I recommend you use a `while` loop. 
```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw 
    game_on = false
  end
end
```
You are using `9.times do ` this will be a problem because if you select the same position 9 times the game ends. Please fix this.
---------------
Make the required changes and send it again for review. :+1: 
If you have any questions reach me on SLACK.
@ Brian Sammit Cruz Rodriguez
HAPPY CODING!!! :smile: 
MB5W4D

**Comment by KabohaJeanMark** [comment]
Hello @BrianSammit. Thanks for the quick feedback and for presenting an alternative option.

Currently, we are validating user input (we test if it is an integer greater than 0 and less than 10). We are not yet tracking the cells that have been marked. Once we start doing that (in the next milestone where we work on the game logic), we will simply add the test to our validation and prompt the user for another input if the cell is already marked.

We are validating inside an inner while loop that breaks only when the input is valid. If we use an outer while loop, we will do exactly the same thing (input validation) and in addition, we will have to add more logic to test if all the cells have been marked.

We will most definitely be doing this in the next milestone which is focused on the game logic. Thanks 👍🏾

File: bin/main.rb [hunk 1]
```diff
@@ -1,7 +1,107 @@
...
+  lines = 6
+
+  name = turn.even? ? player1_name : player2_name
+  puts
+  puts "It's #{name}'s turn"
+  puts
+  puts 'Please select an available cell from the board `- -`'
+  puts
+  cell = user_input.to_i
+  while cell < 1 || cell > 9
```
Comment (pacyL2K19) [inline bin/main.rb:92]: ## To improve 
This works and you are checking the valid move before moving to the next iteration of the `9.times` loop 
A loop in a loop will always seem to be a sign of lower performance, in the next milestone, kindly avoid as much as you can such of behavior, you can, for example, create a separate method for that handling the move of the user and call it just once you need to decide if the move is valid or not

**Review by pacyL2K19 — APPROVED** [review]
# Status: Approved

Hello @KabohaJeanMark and @chasscepts  you made a great job so far, it was a pleasure to read your code, I enjoyed :tada: 
Now you can move to the next milestone
It's always important to consider all the feedback given even when they are optional, kindly consider them as improvement of your project 
Feel free to connect on Slack or [LinkedIn](https://www.linkedin.com/in/pacifique-linjanja/) 
### Happy coding :desktop_computer:

File: bin/main.rb (hunk 1 above)
Comment (KabohaJeanMark) [reply bin/main.rb:92]: Hey 👍🏾  @pacyL2K19   Yes, thank you for the feedback and recommendation for improvement.

What you are saying is true. @chasscepts and I  are refactoring the code for this next milestone using neat helper methods that we are calling for a more modular DRY approach 💪🏾  to handle the validation and checks.

Comment (pacyL2K19) [reply bin/main.rb:92]: Keep going forward, happy coding

**Pull request merged**